scripts/merge_truth_per_tract.py

```diff
@@ -0,0 +1,184 @@
+
+    Parameters
+    ----------
+    input_dir : str
+        Path to the input tract directory
+
+    Optional Parameters
+    ----------------
+    validate : bool, optional (default: False)
+        If true, check the tract column has only one value
+    silent : bool, optional (default: False)
+        If true, turn off most printout.
+    """
+    my_print = (lambda x: None) if silent else print
+
+    my_print("Merging all files in", input_dir)
+    files_to_merge = sorted(os.listdir(input_dir))
+    df_to_merge = []
+    for filename in files_to_merge:
+        my_print("Loading", filename)
+        df = pd.read_parquet(os.path.join(input_dir, filename))
+        m = re.search(r"_hp(\d+)\b", filename)
+        if m is not None:
+            hp = int(m.groups()[0])
+            df["cosmodc2_hp"] = hp
+            df["cosmodc2_id"] = df["id"]
+            df["id"] = df["id"].astype(str)
+        else:
+            df["cosmodc2_hp"] = -1
+            df["cosmodc2_id"] = -1
+        df["is_sn"] = True if filename.startswith("sn_") else False
+        df_to_merge.append(df)
+    df = pd.concat(df_to_merge, ignore_index=True)
+    del df_to_merge
+
+    if validate:
+        assert len(np.unique(df["tract"].values)) == 1
+
+    return df
+
+
+def match_object_with_merged_truth(truth_cat, object_cat, validate=False, silent=False, **kwargs):
+    """Match object catalog with truth catalog
+
+    Parameters
+    ----------
+    truth_cat : str or pd.DataFrame
+        Truth catalog or its path
+    object_cat : str or pd.DataFrame
+        Object catalog or its path
+
+    Optional Parameters
+    ----------------
+    validate : bool, optional (default: False)
+        If true, check the matching is perform properly.
+    silent : bool, optional (default: False)
+        If true, turn off most printout.
+    """
+    my_print = (lambda x: None) if silent else print
+
+    if isinstance(truth_cat, str):
+        my_print("Loading truth catalog from", truth_cat)
+        truth_cat = pd.read_parquet(truth_cat)
+
+    if isinstance(object_cat, str):
+        my_print("Loading object catalog from", object_cat)
+        object_cat = pd.read_parquet(object_cat, columns=["objectId", "ra", "dec"])
+
+    my_print("Performing nearest neighbor match")
+    object_sc = SkyCoord(object_cat["ra"].values, object_cat["dec"].values, unit="deg")
+    truth_sc = SkyCoord(truth_cat["ra"].values, truth_cat["dec"].values, unit="deg")
+    idx, sep, _ = object_sc.match_to_catalog_sky(truth_sc)
+    del object_sc, truth_sc
+
+    my_print("Merging matching info with truth catalog")
+    matched = pd.DataFrame.from_dict({
+        "truth_idx": truth_cat.index.values[idx],
+        "match_sep": sep.arcsec,
+        "match_objectId": object_cat["objectId"]
+    })
+    del idx, sep
+
+    matched.sort_values("match_sep", inplace=True)
+    matched["is_unique_truth_entry"] = ~matched.duplicated("truth_idx", keep="first")
```

Comment:
```is_unique_truth_entry``` is a slightly misleading name. Coming up with a wholly accurate name which isn't very long would be hard, though.

Reply:
I haven't changed this because I cannot come up with a better idea. If you do, let me know :slightly_smiling_face: